From b58c4af95bbd5d76f8b21d43f0f645083c566af5 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Thu, 4 Apr 2024 10:43:03 +0200
Subject: [PATCH 0795/1695] dt-bindings: dma: add st,syscfg-arcr property to
 st,stm32-dma3

'st,syscfg-arcr' property is used to get register offset and register
bitmask within system configuration controller (syscfg), for DMA controller
Address Remapping ENable (AREN) bit.
When this bit is set, it means STM32 DMA3 AXI port address remapping is
enabled, then the 4GB address space accessible by the STM32 DMA3 AXI port
starts at address 0x80000000 instead of 0x00000000.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 5fb46bb65458f21382fa3d335bcd70c348733803 dt-bindings: dma: add st,syscfg-arcr property to st,stm32-dma3]
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Change-Id: I1ad1c9e4bbf19bd010a366487ea725d328f2d179
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/372997
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 .../devicetree/bindings/dma/st,stm32-dma3.yaml       | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/documentation/devicetree/bindings/dma/st,stm32-dma3.yaml b/documentation/devicetree/bindings/dma/st,stm32-dma3.yaml
index d179f5d64..4fa665299 100644
--- a/documentation/devicetree/bindings/dma/st,stm32-dma3.yaml
+++ b/documentation/devicetree/bindings/dma/st,stm32-dma3.yaml
@@ -122,6 +122,18 @@ properties:
     minimum: 1
     maximum: 256
 
+  st,syscfg-arcr:
+    description: |
+      Phandle to system configuration controller. It is used to get the status of the STM32 DMA3 AXI
+      port address remapping. When enabled, the 4GB address space accessible by the STM32 DMA3 AXI
+      port starts at address 0x80000000 instead of 0x00000000.
+    $ref: /schemas/types.yaml#/definitions/phandle-array
+    items:
+      - items:
+          - description: phandle to syscfg
+          - description: register offset within syscfg
+          - description: register bitmask for DMA controller Address Remapping ENable (AREN) bit
+
     properties:
       st,protreg:
         minItems: 1
-- 
2.47.2

